### PR TITLE
Fix large memory use caused by the use of multiprocessing.imap_unordered.

### PR DESCRIPTION
**Pull request by Bodong-Yang**
### Supersede #117

# Introduction
Previously, I introduce `multiprocessing.imap_unordered` to accelerate the `RegularInf` parsing.
But after small scale tests with `memory profiling` and `timeit`, I found that comparing to the time it saves(around 10seconds), it consume 200MB~ extra memory, which is not so efficient.
This PR dips the use of `multiprocessing.imap_unordered`, and also reduce some memory footprint for `RegularInf` class.

# Test and result
For one mainecu and one subecu environment, the memory usage on the mainecu drops from `867M~` to `664M~`.

### CODE REVIEW

**Comment by Bodong-Yang** [comment]
black is broken on version<22.3.0.
https://github.com/psf/black/issues/2964

File: app/ota_client.py [hunk 1]
```diff
@@ -231,74 +231,93 @@ class _BaseInf:
...
 
+    __slots__ = ["mode", "uid", "gid", "_left"]
+
     @staticmethod
     def de_escape(s: str) -> str:
         return s.replace(r"'\''", r"'")
 
     def __init__(self, info: str):
-        match_res: re.Match = self._base_pattern.match(info.strip("\n"))
+        match_res: re.Match = self._base_pattern.match(info.strip())
```
Comment (obi-t4) [inline app/ota_client.py:241]: @Bodong-Yang This strip() strips not only \n but also (=space)?
This should be reverted?

Comment (Bodong-Yang) [reply app/ota_client.py:241]: OK!

Comment (obi-t4) [reply app/ota_client.py:241]: > end with white space?

I think info.strip is not only `end with white space`. Is my understanding correct?

Comment (obi-t4) [reply app/ota_client.py:241]: Ah, I thought "a b c".strip() returns "abc", but it returns "a b c".
But " a b c ".strip() returns "a b c", so I think strip("\n") should be used.

Comment (Bodong-Yang) [reply app/ota_client.py:241]: @obi-t4 Ah I mean it there any case that info string will need to preserved the ending white space? Path itself is wrapped with `'` so I think any white space at the end of info string is useless?

Comment (obi-t4) [reply app/ota_client.py:241]: Yes, you're right! strip() is fine!

Comment (Bodong-Yang) [reply app/ota_client.py:241]: > Ah, I thought "a b c".strip() returns "abc", but it returns "a b c". But " a b c ".strip() returns "a b c", so I think strip("\n") should be used.

Yes, `strip` only strip alway the prefix/suffix ` ` or `\n`

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**